src/lib/components/Map/Map.js

```diff
@@ -181,18 +182,38 @@ class Map extends Component {
   //   }
   // }
   onFeatureClick(e) {
-    const { layerObj, mapId, layersObj, layers } = this.props;
+    const { mapId, layersObj, layers } = this.props;
     const activeLayers = layersObj.map(l => l.id);
     const features = this.map.queryRenderedFeatures(e.point, {
       layers: activeLayers.filter(l => this.map.getLayer(l) !== undefined),
     });
     /** Inestigate why fill layer picks one feature object which is not what we desire */
     const feature = features[0];
     if (!feature) return false;
-    const activeLayerObj = layersObj.find(l => l.id === feature.layer.id);
+    const activeLayerObj = layersObj.find((l) => l.id === feature.layer.id);
+    /**
+     * Todo:
+     * Investigate adjusting zoom to show selected feature
+     */
+    if (feature.layer && feature.layer.id && activeLayerObj[HIGHLIGHT_FILTER_PROPERTY]) {
+      /**
+       * 1. Filter out the selected feature based on period and facility id 
+       (equal to feature period & not equal to facility_id)
+       * 2. Filter in highlight layer features with simillar facility_id
+       * 3. Set Icon opacity to one for the highlight layer 
+       * (This was set to 0 on initial render not to show the icon before it's filtered out)
+       * 4. Filter nutrition-site-live with same reporting_period and different facility_id from selected feature
+       * 5. Move highlight layer to the top (ensure the highlighted layer is not hidden due to mapbox clustering)
+       */
+      this.map.setFilter(activeLayerObj.id, [ALL, ["==", 'reporting_period', feature.properties.reporting_period], ["!=", activeLayerObj.source.join[0], feature.properties.facility_id]]);
+      this.map.setFilter(`${activeLayerObj.id}${HIGHLIGHT}`, [ALL, ["==", activeLayerObj.source.join[0], feature.properties.facility_id]]);
+      this.map.setPaintProperty(`${activeLayerObj.id}${HIGHLIGHT}`, ICON_OPACITY, 1);
+      this.map.moveLayer(`${activeLayerObj.id}${HIGHLIGHT}`);
+    }
```

Comment:
Nice! For number `1`, will the reporting period property always be `reporting_period`?

If so then could it go in `constants.js`? If not, is it available from the layer config?

Reply:
@cKellyDesign Nice catch man. reporting_period is a field in the data. This will change based on various clients. I think adding a property on the layer spec would be better than having it on the constants. what's your take on this ?

Reply:
@kahummer I thought we already had something in the spec defining which field to use 🤔

Reply:
```
{
  ...
  "aggregate": {
        "timeseries": {
            "field": "period"
        }
    }
}
```

Reply:
@cKellyDesign In this case we are building the periods in gisida-private. We don't have the field provide https://github.com/onaio/nutrition-cluster/blob/enhancement-nutrition-sites/public/config/layers/nutrition-sites-live.json#L39

Reply:
@kahummer what happens if we add `aggregate.timeseries.field` in that case? If needed then yeah we can add a new prop, it would be nice if they could be the same but not necessary.

Reply:
I think this would build the normal map  timeseries and seems the periods for this case are not based on the entire layer but on a specific feature. 
> If needed then yeah we can add a new prop, it would be nice if they could be the same but not necessary.
>
:point_up: this is needed